client/opkclient_test.go

```diff
@@ -38,36 +37,38 @@ func TestClient(t *testing.T) {
 		name        string
 		gq          bool
 		signer      bool
-		alg         jwa.KeyAlgorithm
+		signerAlg   jwa.KeyAlgorithm
 		extraClaims map[string]string
 	}{
 		{name: "without GQ", gq: false, signer: false},
 		{name: "with GQ", gq: true, signer: false},
-		{name: "with GQ, with signer", gq: true, signer: true, alg: jwa.RS256},
-		{name: "with GQ, with signer, with empty extraClaims ", gq: true, signer: true, alg: jwa.ES256, extraClaims: map[string]string{}},
-		{name: "with GQ, with signer, with extraClaims", gq: true, signer: true, alg: jwa.ES256, extraClaims: map[string]string{"extra": "yes"}},
+		{name: "with GQ, with signer", gq: true, signer: true, signerAlg: jwa.RS256},
+		{name: "with GQ, with signer, with empty extraClaims", gq: true, signer: true, signerAlg: jwa.ES256, extraClaims: map[string]string{}},
+		{name: "with GQ, with signer, with extraClaims", gq: true, signer: true, signerAlg: jwa.ES256, extraClaims: map[string]string{"extra": "yes"}},
 		{name: "with GQ, with extraClaims", gq: true, signer: false, extraClaims: map[string]string{"extra": "yes", "aaa": "bbb"}},
 	}
 
-	op, err := providers.NewMockOpenIdProvider()
-	require.NoError(t, err, "failed to create mock OpenIdProvider")
+	provider, err := mocks.NewMockOpenIdProvider(t, map[string]any{})
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 
 			var c *client.OpkClient
 			if tc.signer {
-				signer, err := util.GenKeyPair(tc.alg)
+				signer, err := util.GenKeyPair(tc.signerAlg)
 				require.NoError(t, err, tc.name)
-				c, err = client.New(op, client.WithSignGQ(tc.gq), client.WithSigner(signer, tc.alg))
+				c, err = client.New(provider, client.WithSignGQ(tc.gq), client.WithSigner(signer, tc.signerAlg))
 				require.NoError(t, err, tc.name)
 				require.Equal(t, signer, c.GetSigner(), tc.name)
-				require.Equal(t, tc.alg, c.GetAlg(), tc.name)
+				require.Equal(t, tc.signerAlg, c.GetAlg(), tc.name)
 			} else if tc.gq {
-				c, err = client.New(op, client.WithSignGQ(tc.gq))
+				c, err = client.New(provider, client.WithSignGQ(tc.gq))
 				require.NoError(t, err, tc.name)
 			} else {
-				c, err = client.New(op)
+				c, err = client.New(provider)
```

Comment:
I like your rename of signerAlg here as it increases clarity. My original name was a little confusing, is this the alg for the OP or the alg for the signer.

For the same reasons I like you renaming of signerAlg, I find the renaming of op to provider to be confusing since we have OpenIDProviders and CosignerProviders and the client can take both of them.

You know me, I love short names so I prefer op (OpenIDProvider) and cosP (CosignerProvider), but you really must to use a longer name here, can you use a name that makes it clear this is an OpenIDProvider such as `oidcProvider`, `oProvider`. Just something that allows a develop at a glance to know which provider you are referencing.

Reply:
Reading more of the code and I think I see what you are doing here. You have renamed the CosignerProvider to Cosigner so that provider always implies OP. Let me get that some thought. I think you can ignore my above comment for now, I wrote it before I got to code where you make that change.

Reply:
Go really wants you to use single, whole words, a point you've brought up in the past, because it's a lot more readable and explanatory. I will say that I find it confusing to have an oidc provider and a cosigner provider so assigning each a distinct yet descriptive word I think helps with the overall clarity. But I'll wait for your further comments.

Reply:
I think I misunderstood your intention here. I thought you were renaming op to provider everywhere because you were no longer referring to the cosigner as a provider. However it appears the cosigner is still named a cosigner provider.  Given that we use the Op in the client struct and in the constructor to the client `func New(op OpenIdProvider, opts ...ClientOpts) (*OpkClient, error) {` I don't see a justification to renaming and touching all these lines of code in the tests.

This is existing code and changing it is both out of scope for this PR and reduces the consistency of variable names in this code base.